### PR TITLE
Change some integer types to u8

### PR DESCRIPTION
**Pull request by vbkaisetsu**
This branch changes the type of the following options to `u8`:
* `charw`, `charn`, `typew`, `typen`, `dictn`

and uses `from()` for static casting instead of `as` in some places.

### CODE REVIEW

File: vaporetto/src/tag_trainer.rs [hunk 1]
```diff
@@ -76,31 +76,35 @@ impl<'a> TagTrainer<'a> {
...
+        let quantize_multiplier = weight_max / f64::from((1 << (QUANTIZE_BIT_DEPTH - 1)) - 1);
 
         let mut class_info = vec![];
 
         for i in 0..self.tag_ids.len() {
             class_info.push(TagClassInfo {
                 name: self.tag_ids.keys()[model.labels()[i] as usize].clone(),
-                bias: (model.label_bias(i as i32) / quantize_multiplier) as i32,
+                bias: unsafe {
+                    (model.label_bias(i32::try_from(i)?) / quantize_multiplier).to_int_unchecked()
```
Comment (kampersanda) [inline vaporetto/src/tag_trainer.rs:99]: For safety, `quantize_multiplier` must not be zero.
Could you add assertion `assert_ne!(quantize_multiplier, 0.)` before the for-loop?

File: vaporetto/src/trainer.rs [hunk 2]
```diff
@@ -151,9 +151,9 @@ impl From<SolverType> for liblinear::SolverType {
 pub struct Trainer<'a> {
     dictionary: Vec<String>,
     example_generator: BoundaryExampleGenerator,
-    char_window_size: usize,
-    type_window_size: usize,
-    dict_max_word_size: usize,
+    char_window_size: u8,
+    type_window_size: u8,
+    dict_max_word_size: u8,
```
Comment (kampersanda) [inline vaporetto/src/trainer.rs:156]: `u16`? (as I commented before)

File: vaporetto/src/trainer.rs [hunk 3]
```diff
@@ -285,33 +285,35 @@ impl<'a> Trainer<'a> {
...
+            let weight = model.feature_coefficient(i32::try_from(fid)?, wb_idx).abs();
             if weight > weight_max {
                 weight_max = weight;
             }
         }
-        let quantize_multiplier = weight_max / ((1 << (QUANTIZE_BIT_DEPTH - 1)) - 1) as f64;
+        let quantize_multiplier = weight_max / f64::from((1 << (QUANTIZE_BIT_DEPTH - 1)) - 1);
 
-        let bias = (bias / quantize_multiplier) as i32;
+        let bias = unsafe { (bias / quantize_multiplier).to_int_unchecked::<i32>() };
```
Comment (kampersanda) [inline vaporetto/src/trainer.rs:312]: Could you add assertion `assert_ne!(quantize_multiplier, 0.)` before this line?

File: vaporetto/src/feature.rs [hunk 4]
```diff
@@ -66,22 +66,22 @@ pub struct BoundaryExample<'a> {
...
-    type_ngram_size: usize,
-    char_window_size: usize,
-    type_window_size: usize,
+    char_ngram_size: u8,
+    type_ngram_size: u8,
+    char_window_size: u8,
+    type_window_size: u8,
     dict_ac: Option<DoubleArrayAhoCorasick>,
-    dict_max_word_size: usize,
+    dict_max_word_size: u8,
```
[the remark this answers is not in the document]
Comment (vbkaisetsu) [reply vaporetto/src/feature.rs:74]: I think u8 is enough to specify the parameter.
In addition, all  u8 can be convert to i16 correctly, but u16 will be wrapped. (e.g. `u16::MAX as i16 == -1`

[the remark this answers is not in the document]
Comment (kampersanda) [reply vaporetto/src/feature.rs:74]: Okey.

**Review by kampersanda — APPROVED** [review]

**Pull request merged**